src/sentry/search/events/builder.py

```diff
@@ -38,28 +82,69 @@ def __init__(
         turbo: bool = False,
         sample_rate: Optional[float] = None,
     ):
-        super().__init__(dataset, params, auto_fields, functions_acl)
+        self.dataset = dataset
+        self.params = params
+        self.auto_fields = auto_fields
+        self.functions_acl = set() if functions_acl is None else functions_acl
+        self.equation_config: Optional[Dict[str, bool]] = {}
 
-        self.auto_aggregations = auto_aggregations
+        # Function is a subclass of CurriedFunction
+        self.where: List[WhereType] = []
+        # The list of aggregates to be selected
+        self.aggregates: List[CurriedFunction] = []
+        self.columns: List[SelectType] = []
+        self.orderby: List[OrderBy] = []
+        self.projects_to_filter: Set[int] = set()
+        self.function_alias_map: Dict[str, FunctionDetails] = {}
 
+        self.auto_aggregations = auto_aggregations
         self.limit = None if limit is None else Limit(limit)
         self.offset = None if offset is None else Offset(offset)
-
-        self.limitby = self.resolve_limitby(limitby)
         self.turbo = Turbo(turbo)
         self.sample_rate = sample_rate
 
+        self.resolve_column_name = resolve_column(self.dataset)
+
+        (
+            self.field_alias_converter,
+            self.function_converter,
+            self.search_filter_converter,
+        ) = self.load_config()
+
+        self.limitby = self.resolve_limitby(limitby)
         self.where, self.having = self.resolve_conditions(
             query, use_aggregate_conditions=use_aggregate_conditions
         )
-
         # params depends on parse_query, and conditions being resolved first since there may be projects in conditions
         self.where += self.resolve_params()
-
         self.columns = self.resolve_select(selected_columns, equations)
         self.orderby = self.resolve_orderby(orderby)
         self.array_join = None if array_join is None else [self.resolve_column(array_join)]
 
+    def load_config(
+        self,
+    ) -> Tuple[
+        Mapping[str, Callable[[str], SelectType]],
+        Mapping[str, SnQLFunction],
+        Mapping[str, Callable[[SearchFilter], Optional[WhereType]]],
+    ]:
+        from sentry.search.events.dataset import DiscoverDatasetConfig
+
+        if self.dataset in [Dataset.Discover, Dataset.Transactions, Dataset.Events]:
+            self.config = DiscoverDatasetConfig(self)
+        else:
+            raise NotImplementedError(f"Data Set configuration not found for {self.dataset}.")
+
+        field_alias_converter = self.config.field_alias_converter
+
+        function_converter = self.config.function_converter
+        for alias, name in FUNCTION_ALIASES.items():
+            function_converter[alias] = function_converter[name].alias_as(alias)
```

Comment:
Function aliases should probably live within the config as it's dataset specific.